Configs: fix `TS2629` & `no-class-assign` duplicate

### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22preset+config+change%22+) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Description

I followed [the official instrusctions](https://typescript-eslint.io/getting-started/), but still stumbled upon this ESL rule enabled by default

### Impacted Configurations

`recommended`

### Additional Info

`eslint.config.mjs`:
```js
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
	eslint.configs.recommended,
	...tseslint.configs.recommended,
);
```

`tsconfig.json`:
```json
{
"compilerOptions": {
	"target": "ES2023",
	"skipLibCheck": true
},
"exclude": [ "node_modules" ]
}
```

`package.json`:
```json
{
"name": "tsesl-min-repro",
"version": "0.0.0",
"main": "main.js",
"devDependencies": {
	"@eslint/js": "^9.13.0",
	"@types/eslint__js": "^8.42.3",
	"eslint": "^9.13.0",
	"typescript": "^5.6.3",
	"typescript-eslint": "^8.12.1"
}
}
```

`main.ts`:
```ts
class c { }
c = null
```

`npx eslint .`:
```
/home/rudxain/tsesl-min-repro/main.ts
  2:1  error  'c' is a class                          no-class-assign
  2:1  error  'c' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 2 problems (2 errors, 0 warnings)
```

`npx tsc --noEmit`:
```
main.ts(2,1): error TS2629: Cannot assign to 'c' because it is a class.
```

--- a/packages/eslint-plugin/src/configs/eslint-recommended-raw.ts
+++ b/packages/eslint-plugin/src/configs/eslint-recommended-raw.ts
@@ -22,6 +22,7 @@ const config = (
   rules: {
     'constructor-super': 'off', // ts(2335) & ts(2377)
     'getter-return': 'off', // ts(2378)
+    'no-class-assign': 'off', // ts(2629)
     'no-const-assign': 'off', // ts(2588)
     'no-dupe-args': 'off', // ts(2300)
     'no-dupe-class-members': 'off', // ts(2393) & ts(2300)

